### PR TITLE
supplement season data with source

### PR DESCRIPTION
**Pull request by delphiactual**
by way of https://github.com/delphiactual/d2-additional-info/pull/16 in association with @sundevour 
compare with #3949 

supplement season data via source, filter out the following itemCategoryHashes:
```
  null,
  16, // Quest Steps
  18, // Currencies
  34, // Engrams
  40, // Material
  53, // Quests
  58, // Clan Banner
  268598612, // Packages
  303512563, // Bonus Mods
  444756050, // Weapon Mods: Bow strings
  945330047, // Weapon Mods: Gameplay
  1334054322, // Weapon Mods: Batteries
  1449602859, // Ghost Mods
  1576735337, // Clan Banner: Perks
  1709863189, // Weapon Mods: Sword Blades
  1784235469, // Bounties
  2005599723, // Prophecy Offerings
  2076918099, // Weapon Mods: Launch Tubes
  2150402250, // Gags
  2237038328, // Weapon Mods: Intrinsic
  2250046497, // Prophecy Tablets
  2253669532, // Treasure Maps
  2411768833, // Weapon Mods: Scopes
  2423200735, // Item Sets
  3055157023, // Weapon Mods: Stocks
  3072652064, // Weapon Mods: Sword Guards
  3085181971, // Weapon Mods: Barrels
  3109687656, // Dummies
  3360831066, // Weapon Mods: Arrows
  3708671066, // Weapon Mods: Frames
  3836367751, // Weapon Mods: Grips
  3866509906, // Weapon Mods: Sights
  4184407433 // Weapon Mods: Magazines
```

d2-seasons.json reduced by ~3416~ ~4251~ 5304 line items

### CODE REVIEW

File: src/app/inventory/store/d2-item-factory.service.ts [hunk 1]
```diff
@@ -648,7 +654,23 @@ function getClassTypeNameLocalized(defs: D2ManifestDefinitions, type: DestinyCla
...
   }
 }
+function getSeason(itemHash, source, categoryHashes, typeName) {
+  let sourceToSeason;
+  if (
+    D2SeasonToSource.categoryBlacklist.filter((hash) => categoryHashes.includes(hash)).length ||
+    !categoryHashes.length ||
+    typeName === 'Unknown'
+  ) {
+    return -1;
```
Comment (delphiactual) [inline src/app/inventory/store/d2-item-factory.service.ts]: `season:-1` can be used for debugging items that are not assigned a season

File: src/app/inventory/store/d2-item-factory.service.ts [hunk 2]
```diff
@@ -648,7 +649,28 @@ function getClassTypeNameLocalized(defs: D2ManifestDefinitions, type: DestinyCla
...
+  if (
+    D2SeasonToSource.categoryBlacklist.filter((itemHash) =>
+      item.itemCategoryHashes.includes(itemHash)
+    ).length ||
+    !item.itemCategoryHashes.length ||
+    item.typeName === 'Unknown'
+  ) {
+    return -1;
+  }
+  Object.keys(D2SeasonToSource.seasons).forEach((season) => {
```
Comment (bhollis) [inline src/app/inventory/store/d2-item-factory.service.ts]: Not important, but you could use a standard `for..of` loop here and early return.

File: src/app/inventory/store/d2-item-factory.service.ts [hunk 3]
```diff
@@ -648,7 +649,28 @@ function getClassTypeNameLocalized(defs: D2ManifestDefinitions, type: DestinyCla
...
+    if (D2SeasonToSource.seasons[season].includes(item.source)) {
```
Comment (bhollis) [inline src/app/inventory/store/d2-item-factory.service.ts]: Would you mind documenting the `source` property (and the other properties like it) in `item-types.ts`? I'm not actually super sure what "source" means and so it's a bit hard to follow what the mappings in `D2SeasonToSource` are.

File: src/app/inventory/store/d2-item-factory.service.ts (hunk 2 above)
Comment (delphiactual) [reply src/app/inventory/store/d2-item-factory.service.ts]: `for..of`  says `D2SeasonToSource.seasons` is not iterable

and early out causes a bunch of items to be misidentified, I'm not sure why.
